diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt b/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt
index 0d7947e..48b283e 100644
--- a/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt
@@ -275,7 +275,7 @@ open class PermissionsService(val context: Context) : InternalModule, Permission
       if (requestCode == PERMISSIONS_REQUEST) {
         synchronized(this@PermissionsService) {
           val currentListener = requireNotNull(mCurrentPermissionListener)
-          currentListener.onResult(parseNativeResult(receivePermissions, grantResults))
+          currentListener.onResult(parseNativeResult(receivePermissions ?: emptyArray(), grantResults))
           mCurrentPermissionListener = null
 
           mPendingPermissionCalls.poll()?.let { pendingCall ->
